bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +66,189 @@
+    private static final Duration REINITIALIZE_DELAY_HOURS_LIMIT_EXCEEDED = Duration.ofHours(24).plusSeconds(10);
+
+    // localisation constants
+    private static final String RECONNECT_MSG_KEY = "accounthandler.waiting-until-to-reconnect";
 
+    // assets
+    private @Nullable GardenaDeviceDiscoveryService discoveryService;
     private @Nullable GardenaSmart gardenaSmart;
-    private HttpClientFactory httpClientFactory;
-    private WebSocketFactory webSocketFactory;
+    private final HttpClientFactory httpClientFactory;
+    private final WebSocketFactory webSocketFactory;
+    private final TimeZoneProvider timeZoneProvider;
+    private final Bundle bundle;
+    private final TranslationProvider i18nProvider;
+    private final LocaleProvider localeProvider;
 
-    public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory,
-            WebSocketFactory webSocketFactory) {
+    // re- initialisation stuff
+    private final Object reInitializationCodeLock = new Object();
+    private @Nullable ScheduledFuture<?> reInitializationTask;
+    private @Nullable Instant apiCallSuppressionUntil;
+
+    public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory, WebSocketFactory webSocketFactory,
+            TimeZoneProvider timeZoneProvider, Bundle bundle, TranslationProvider i18nProvider,
+            LocaleProvider localeProvider) {
         super(bridge);
         this.httpClientFactory = httpClientFactory;
         this.webSocketFactory = webSocketFactory;
+        this.timeZoneProvider = timeZoneProvider;
+        this.bundle = bundle;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+    }
+
+    /**
+     * Load the api call suppression until property.
+     */
+    private void loadApiCallSuppressionUntil() {
+        try {
+            Map<String, String> properties = getThing().getProperties();
+            apiCallSuppressionUntil = Instant
+                    .parse(properties.getOrDefault(GardenaBindingConstants.API_CALL_SUPPRESSION_UNTIL, ""));
+        } catch (DateTimeParseException e) {
+            apiCallSuppressionUntil = null;
+        }
+    }
+
+    /**
+     * Get the duration remaining until the end of the api call suppression window, or Duration.ZERO if we are outside
+     * the call suppression window.
+     *
+     * @return the duration until the end of the suppression window, or zero.
+     */
+    private Duration apiCallSuppressionDelay() {
+        Instant now = Instant.now();
+        Instant until = apiCallSuppressionUntil;
+        return (until != null) && now.isBefore(until) ? Duration.between(now, until) : Duration.ZERO;
+    }
+
+    /**
+     * Updates the time when api call suppression ends to now() plus the given delay. If delay is zero or negative, the
+     * suppression time is nulled. Saves the value as a property to ensure consistent behaviour across restarts.
```

Comment:
Please be aware that you cannot rely on properties to be persisted:
https://www.openhab.org/docs/developer/bindings/#updating-thing-properties

I don't know the details, but as far as I remember properties are not persisted across openHAB restarts for unmanaged things. This can easily be verified by testing it.

Reply:
Yes I am aware of the limitations of this. As far as I can tell there are several 'levels' of restart -- namely 1) disable/enable the binding, 2) reload the JAR, 3) restart OH, 4) reboot the PC. There seems to be no perfect way to persist something like this across all such 'levels'. The solution that I am using does work across some of these however..